File: output/LLaMa/codenet/Python/Java/s814885508.java
import java.util.InputMismatchException;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        try {
            int X = sc.nextInt();
            int K = sc.nextInt();
            int D = sc.nextInt();

            int r = X / D;
            if (Math.abs(r) > K) {
                System.out.println(Math.abs(X) - K * D);
                System.exit(0);
            }
            int a = X - r * D;
            if ((K - r) % 2 == 0) {
                System.out.println(a);
            } else {
                System.out.println(Math.abs(D - a));
            }
        } catch (InputMismatchException e) {
            System.out.println("Error: Invalid input");
        }
    }
}

File: dataset_reduzido/codenet/Python/Code/s814885508.py

def main():
    X, K, D = map(int, input().split())

    r = X // D
    if abs(r) > K:
        print(abs(X) - K * D)
        exit()
    a = X - r * D
    if (K - r) % 2 == 0:
        print(a)
    else:
        print(abs(D - a))
    


if __name__ == '__main__':
    main()